Project/Src/Gui/TextAreaDragDropHandler.cs

```diff
@@ -93,24 +96,35 @@ protected void OnDragDrop(object sender, DragEventArgs e)
                     var offset = textArea.Caret.Offset;
                     if (textArea.IsReadOnly(offset))
                         return;
-                    if (e.Data.GetDataPresent(typeof(DefaultSelection)))
+
+                    try
                     {
-                        var sel = (ISelection)e.Data.GetData(typeof(DefaultSelection));
-                        if (sel.ContainsPosition(textArea.Caret.Position))
-                            return;
-                        if (GetDragDropEffect(e) == DragDropEffects.Move)
+                        if (e.Data.GetDataPresent(typeof(DefaultSelection)))
                         {
-                            if (SelectionManager.SelectionIsReadOnly(textArea.Document, sel))
+                            var sel = (ISelection)e.Data.GetData("ICSharpCode.TextEditor.Document.DefaultSelection");
```

Comment:
Why?

Reply:
Sorry I think this "GetData using type name string" is a remnant from debugging and can be changed back to use `typeof(DefaultSelection)`.